crates/net/common/src/network_io_meter.rs

```diff
@@ -165,6 +199,18 @@ impl HasRemoteAddr for MeteredStream<TcpStream> {
     }
 }
 
+/// Allows exporting metrics for a type that contains a nested [`MeteredStream`]
+pub trait MeterableStream {
```

Comment:
Not sure about this name, to me it seems to suggest that it is impl for streams that can be turned into a `MeteredStream`. The actual usage seems to be to 1) call `expose_metrics` on an inner `MeteredStream` instance and 2) assign a `NetworkIOMetrics` to `MeteredStream` if called directly on `MeteredStream`

For 1) is there any reason we could not just `impl AsMut<MeteredStream> for ...`? That way you could do `.as_mut().set_meter(NetworkIOMeter::new())`.